src/main/java/coydir/logic/commands/AddLeaveCommand.java

```diff
@@ -0,0 +1,82 @@
+package coydir.logic.commands;
+
+import static coydir.logic.parser.CliSyntax.PREFIX_ENDDATE;
+import static coydir.logic.parser.CliSyntax.PREFIX_ID;
+import static coydir.logic.parser.CliSyntax.PREFIX_STARTDATE;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import coydir.commons.core.Messages;
+import coydir.logic.commands.exceptions.CommandException;
+import coydir.model.Model;
+import coydir.model.person.EmployeeId;
+import coydir.model.person.Leave;
+import coydir.model.person.Person;
+
+
+/**
+ * Add leave for employee
+ */
+public class AddLeaveCommand extends Command {
+
+    public static final String COMMAND_WORD = "addleave";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Adds a leave period to an employee.\n"
+            + "Parameters: "
+            + PREFIX_ID + "ID "
+            + PREFIX_STARTDATE + "START DATE "
+            + PREFIX_ENDDATE + "END DATE "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_ID + "1 "
+            + PREFIX_STARTDATE + "01-01-22 "
+            + PREFIX_ENDDATE + "06-01-22 ";
+
+    public static final String MESSAGE_LEAVE_ADDED_SUCCESS = "Leave added successfully for %1$s";
+    public static final String MESSAGE_DUPLICATE_LEAVE = "This leave period already exists";
+    public static final String MESSAGE_OVERLAPPING_LEAVE = "You cannot have overlapping leaves";
```

Comment:
Not sure if 2nd person makes sense actually, maybe can just say something like "Overlapping leave periods are not allowed."